Ability to reset personal lockers

ganglyalexander opened the pull request:
Crow recommended that it be added to port 6, so here ya go

**Description**
Right clicking personal lockers/cabinets will grant you the option to reset the ownership of it, the locker/cabinet has to be open in order for the lock to be reset. 
The description of the locker changes if someone were to examine it and the next person to swipe their ID will have ownership of the locker.

ganglyalexander committed:
Ability to reset personal lockers

Gives the player the option to reset the ownership of lockers if and
ONLY IF the locker is open

Addition
-reset the ownership of a personal locker

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,22 @@
+
+	if(opened)
+		if(src.broken)
+			usr << "<span class='danger'>It appears to be broken.</span>"
```
Maybe be a little more descriptive here about what's broken exactly.

TalkingCactus replied:
Also you don't need to call `src` in most of these var checks and such. `if(broken)` is the same as `if(src.broken)`, only faster.

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,22 @@
+		else
+			registered_name = null
+			usr << "<span class='danger'>You successfully reset the lock.</span>"
+			src.desc = "The lock appears to be reset"
```
desc = initial(desc)

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,22 @@
+			usr << "<span class='danger'>You successfully reset the lock.</span>"
+			src.desc = "The lock appears to be reset"
+			add_fingerprint(usr)
+			return
```
The less `return`s the better, only use them when you need to

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,22 @@
+			add_fingerprint(usr)
+			return
+	else
+		usr << "<span class='danger'>The locker must be open!</span>"
```
`"<span class='danger'>\the [src.name] must be open!</span>"`

ganglyalexander committed:
Ability to reset lockers FIX

The recommand parts have been added to the code.
Thank you very much Cactus for looking at it

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,21 @@
+			desc = initial(desc)
+			add_fingerprint(usr)
+	else
+		usr << "<span class='danger'>The [src.name] must be open!</span>"
```
Did `\the [src.name] must be open!` not work here? `\the` will choose whether or not to place a \the there according to the surrounding grammar, so it's modular. It helps when mappers or coders give lockers proper names (capitalized) and the game assumes they're proper nouns.

TalkingCactus replied:
Alternatively you can just do `[src.name]` without `\the` and it will automatically place a `the` there if it's needed.

TalkingCactus replied:
I need this fixed before I can merge. @ganglyalexander

ganglyalexander replied:
Sorry about the long return in response, had some RL things that needed to be taken care of.

ganglyalexander replied:
hm... The alternative method doesn't seem to fit that well... Trying to reset the locker gives me this "personal closet must be open!"

ganglyalexander replied:
blegh, I need to read more carefully

ganglyalexander committed:
Grammar Fix

Added a \ to the beginning of The for grammar reasons

TalkingCactus commented on code/game/objects/structures/crates_lockers/closets/secure/personal.dm:
```diff
@@ -53,4 +53,21 @@
+			desc = initial(desc)
+			add_fingerprint(usr)
+	else
+		usr << "<span class='danger'>\The [src.name] must be open!</span>"
```
`\The` -> `\the`

TalkingCactus replied:
or just remove the `The` and `\` completely

ganglyalexander committed:
\The changed to \the

\The changed to \the
